psi4_FFI_to_strain.py

```diff
@@ -0,0 +1,426 @@
+"""
+This module processes gravitational wave data from numerical relativity simulations.
+
+The module reads a directory of ASCII files of various ell values of waveform data, computes phase and amplitude data,
+calculates a minimum data frequency using a quadratic fit of the monotonic phase of  ell2 m2 data,
+and uses a Fast Fourier Transform to compute the second time integral of the waveform, (the strain).
+The module computes a second derivative of the result to check against the original data.
+
+The phase and amplitude, the second integral, and the twice integrated-twice differentiated data is saved to txt files.
+The primary function returns the second integral data as a numpy array with the various ell-values.
+
+Author: Zachariah B. Etienne
+        zachetie **at** gmail **dot* com
+"""
+
+import sys
+import os
+from typing import List, Tuple
+import numpy as np
+from numpy.typing import NDArray
+from scipy.optimize import curve_fit
+
+
+INPUT_DIR = "../BH_VIS/data/r100" # changeable in animation_main.py
```

Comment:
Please, no globals. These should be passed from the calling function.

Reply:
New commit, removed all globals, placed sys.argv and default arguments instead (ell_min = 2 remains hard coded, and cutoff_factor should maybe be changed to 1 from 0.75)